configure.ac

```diff
@@ -357,7 +357,7 @@ dnl Some systems (OpenServer 5) dislike -lsocket -lnsl, so we try to avoid -lnsl
 dnl checks, if we already have the functions which are usually in libnsl. Also,
 dnl uClibc will bark at linking with glibc's libnsl.
 
-PHP_CHECK_FUNC(socket, socket, network)
+AC_SEARCH_LIBS([socket], [socket network])
```

Comment:
would it make sense to pair up socket/socketpair ? but then also gethostname/gethostbyaddr ?

Reply:
Yes, definitely. But issue with these next ones is that I'm not sure yet how and if this can be really totally simplified :D For example, this socketpair will need a different check and it has PHP's own implementation on Windows. And using it conditionally still makes sense. Something like this:

```m4
AC_CHECK_FUNCS([socketpair], [AC_SEARCH_LIBS([socketpair], [socket network],
  [AC_DEFINE([HAVE_SOCKETPAIR], [1])])])
```

And that -lnsl and -lsocket combination and I need to recheck this most likely totally outdated comment what's up with that (because this isn't relevant anymore):

```
dnl Some systems (OpenServer 5) dislike -lsocket -lnsl, so we try to avoid -lnsl
dnl checks, if we already have the functions which are usually in libnsl. Also,
dnl uClibc will bark at linking with glibc's libnsl.
```

From what I'm currently observing is that more checks of these special cases, the better. If one day (totally hypothetically) the socket function won't be used in the code anymore, also the check for library required for socket can be removed without worrying if it depends on other functions.

Reply:
alrighty :)